src/python/pants/backend/python/providers/pyenv/rules_integration_test.py

```diff
@@ -0,0 +1,122 @@
+from pants.backend.python.dependency_inference import rules as dependency_inference_rules
+from pants.backend.python.goals.run_python_source import PythonSourceFieldSet
+from pants.backend.python.goals.run_python_source import rules as run_rules
+from pants.backend.python.providers.pyenv.rules import rules as pyenv_rules
+from pants.backend.python.target_types import PythonSourcesGeneratorTarget
+from pants.build_graph.address import Address
+from pants.core.goals.run import RunRequest
+from pants.engine.process import InteractiveProcess
+from pants.engine.rules import QueryRule
+from pants.engine.target import Target
+from pants.testutil.rule_runner import RuleRunner, mock_console
+
+
+@pytest.fixture
+def rule_runner() -> RuleRunner:
+    return RuleRunner(
+        rules=[
+            *run_rules(),
+            *pyenv_rules(),
+            *dependency_inference_rules.rules(),
+            *target_types_rules.rules(),
+            QueryRule(RunRequest, (PythonSourceFieldSet,)),
+        ],
+        target_types=[
+            PythonSourcesGeneratorTarget,
+        ],
+    )
+
+
+def run_run_request(
+    rule_runner: RuleRunner,
+    target: Target,
+) -> Tuple[int, str, str]:
+    args = [
+        "--backend-packages=['pants.backend.python', 'pants.backend.python.providers.pyenv']",
+        "--source-root-patterns=['src']",
+    ]
+    rule_runner.set_options(args, env_inherit={"PATH", "PYENV_ROOT", "HOME"})
+    run_request = rule_runner.request(RunRequest, [PythonSourceFieldSet.create(target)])
+    run_process = InteractiveProcess(
+        argv=run_request.args,
+        env=run_request.extra_env,
+        input_digest=run_request.digest,
+        run_in_workspace=True,
+        immutable_input_digests=run_request.immutable_input_digests,
+        append_only_caches=run_request.append_only_caches,
+    )
+    with mock_console(rule_runner.options_bootstrapper) as mocked_console:
+        result = rule_runner.run_interactive_process(run_process)
+        stdout = mocked_console[1].get_stdout()
+        stderr = mocked_console[1].get_stderr()
+
+    return result.exit_code, stdout.strip(), stderr.strip()
+
+
+@pytest.mark.parametrize("py_version", ["==2.7.*", "==3.9.*"])
+def test_using_pyenv(rule_runner, py_version):
+    rule_runner.write_files(
+        {
+            "src/app.py": dedent(
+                """\
+                import os.path
+                import sysconfig
+
+                print(sysconfig.get_config_var("prefix"))
+                """
+            ),
+            "src/BUILD": f"python_sources(interpreter_constraints=['{py_version}'])",
+        }
+    )
+
+    target = rule_runner.get_target(Address("src", relative_file_path="app.py"))
+    _1, stdout, _2 = run_run_request(rule_runner, target)
+    named_caches_dir = (
+        rule_runner.options_bootstrapper.bootstrap_options.for_global_scope().named_caches_dir
+    )
+    assert stdout.startswith(f"{named_caches_dir}/pyenv")
```

Comment:
I was hoping/expecting to see an assert that the actual Python used matches the constraint given.

Reply:
That's a good idea